src/rdkafka_conf.c

```diff
@@ -625,6 +625,15 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
 	_RK(ssl.keystore_password),
 	"Client's keystore (PKCS#12) password."
 	},
+    { _RK_GLOBAL, "ssl.certificate.verify_cb", _RK_C_PTR,
+    _RK(ssl.ssl_cert_verify_cb),
+    "Verify the broker certificate callback."
+    },
+    { _RK_GLOBAL, "ssl.certificate.retrieve_cb", _RK_C_PTR,
```

Comment:
@edenhill 

The only problem I see with that approach is that the memory management is a little more complicated from the callers point of view.  They would essentially need to keep the memory around during the entire lifetime of the configuration instance.  Currently once it is sent to OpenSSL it can safely be freed.  However I am ok with either approach since it is only a single buffer that needs to be maintained.

Reply:
Or if we copy it into the config as a buffer not the pointer that would be ok too.  Since it is a public key there is no need to clear the buffer when freeing the memory.

Reply:
Yeah, all config values are copied, so should be fine.

Reply:
@edenhill Are you going to make this change or do you want me to do this and push new changes?

Reply:
@noahdav Thanks, but I'm on it already!

Reply:
@edenhill I looked at this again and tried to see why I did not do just make it a property originally.  The retrieve callback is actually called 3 times.  Once for public key, once for private key, and once for private key password.  Are you planning on making all of those properties?  If so my only objection is that the private key and private key password need to be securely cleared from memory after use since it is sensitive data.  On windows we call SecureZeroMemory to make sure the memory is cleared.  I am assuming some equivalent API on Linux/Unix exists or you could use memset_s with a memory barrier to make sure these are securely wiped before we free the memory.

Reply:
Good and timely input! I'm just now adding functionality to do exactly this for sensitive configuration properties.
Thanks for the hint on SecureZeroMemory :+1:

By "after use", I'm assuming you mean after the rd_kafka_t instance is destroyed, right?

Reply:
Currently I was managing this in my consumer/producer wrapper object after the connection was established.  If you want to move this to before we destroy the instance that is fine too.

Are you planning on having all 3 properties (public key, private key, and private key password) instead of the single callback?  I thought the callback was simpler to have this all in one place since the caller is already implementing the validate callback anyhow.  Also it allowed the caller to implement all three of these since it is not valid to have some of them implemented.  Either way is fine with me as long as we have this same functionality.

Reply:
I'm adding `ssl.certificate.pem` (client's public key) and `ssl.key.pem` (client's private key), taking a PEM-formatted string.
`ssl.key.password` (private key PEM's password, optional) remains unchanged.